File: imports-frontend-entities/src/entities/inspection_override.js
const handler = require('./base/handler')

module.exports = class InspectionOverride {

  constructor(obj) {

    if (!obj) {
      obj = {}
    }

    this.originalDecision = obj.originalDecision
    this.overriddenOn = obj.overriddenOn
    this.overriddenBy = obj.overriddenBy
    this.overrideReason = obj.overrideReason
    this.overrideReasonOther = obj.overrideReasonOther()

    return Object.seal(new Proxy(this, handler))
  }
}

File: notification-schema-java/src/main/java/uk/gov/defra/tracesx/notificationschema/representation/InspectionOverride.java
package uk.gov.defra.tracesx.notificationschema.representation;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import java.time.LocalDateTime;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import uk.gov.defra.tracesx.notificationschema.representation.enumeration.InspectionRequired;
import uk.gov.defra.tracesx.notificationschema.representation.enumeration.OverrideReason;
import uk.gov.defra.tracesx.notificationschema.representation.serialisation.IsoOffsetDateTimeDeserializer;
import uk.gov.defra.tracesx.notificationschema.representation.serialisation.IsoOffsetDateTimeSerializer;

@Builder
@Data
@JsonInclude(Include.NON_EMPTY)
@NoArgsConstructor(access = AccessLevel.PUBLIC)
@AllArgsConstructor(access = AccessLevel.PUBLIC)
public class InspectionOverride {

  private InspectionRequired originalDecision;

  @JsonSerialize(using = IsoOffsetDateTimeSerializer.class)
  @JsonDeserialize(using = IsoOffsetDateTimeDeserializer.class)
  private LocalDateTime overriddenOn;

  private UserInformation overriddenBy;
  private OverrideReason overrideReason;
  private String overrideReasonOther;

}
